server/cluster/cluster.go

```diff
@@ -258,6 +258,7 @@ func (c *RaftCluster) InitCluster(
 
 // Start starts a cluster.
 func (c *RaftCluster) Start(s Server) error {
+	// This is the most common case, so we use a fast path to avoid the lock.
```

Comment:
How about just keeping one check? We use atomic to reduce the cost, but in some cases, we need to check twice, which cost more.

Reply:
+1. May only keep the lock way.

Reply:
I'm fine with lock way, because this isn't high frequent operation.